lambdas/service/app.py

```diff
@@ -1524,6 +1528,27 @@ def handle_manifest_generation_request() -> Tuple[int, Manifest]:
                     catalog=IndexName.validate_catalog_name,
                     filters=str,
                     token=str)
+
+    def amend_manifest_location(manifest: Manifest,
```

Comment:
This still strikes me as too complicated. The fact that this both has arguments AND accesses in-scope local variable makes this very hard to understand.